src/mongo/client/dbclient.cpp

```diff
@@ -1066,15 +1070,175 @@ namespace mongo {
         return runCommand(db.c_str(), b.done(), *info);
     }
 
-    bool DBClientWithCommands::copyDatabase(const string &fromdb, const string &todb, const string &fromhost, BSONObj *info) {
+    string DBClientWithCommands::_authKeyCopyDB(const string &username,
+                                                const string &password,
+                                                const string &nonce)
+    {
+        md5digest d;
+        string passwordDigest = createPasswordDigest( username, password );
+        {
+            md5_state_t st;
+            md5_init(&st);
+            md5_append(&st, (const md5_byte_t *) nonce.c_str(), nonce.size() );
```

Comment:
Can we avoid the C-style cast? I think a static_cast<md5_byte_t*> should work since md5_byte_t is just unsigned char, if not we can use reinterpret_cast.